### DISCUSSION TITLE
Stop sprite from scrolling off screen

### DISCUSSION THREAD

**Discussion by whogate** [Bodmer/TFT_eSPI#2971] [Q&A - Sprites] [answered]
I have a 240 x 240 pixel round display and I'm trying to scroll a 480 x 240 sprite from -240 (off the display) to 0, essentially just taking what's seen on the right half of the sprite and scrolling to see what's on the left half of the sprite. I got it to scroll but it just keeps scrolling past the edge of the sprite. Slide.h is just the byte array for the image I'm pushing into the sprite.

Another thing is that I'd like to have it do this once during the code like an initialization screen. I'm at a loss...

#include <TFT_eSPI.h>
#include "Slide.h"

TFT_eSPI tft = TFT_eSPI();

TFT_eSprite slide = TFT_eSprite(&tft);

void setup() {
  tft.init();

  tft.setRotation(0);
  tft.fillScreen(TFT_BLACK);

  slide.setSwapBytes(true);
  createSlide();
  slide.setScrollRect(0, 0, 480, 240, TFT_WHITE);
}

void loop() {
  slide.pushSprite(-240, 0, TFT_TRANSPARENT);
  slide.scroll(2);
}

void createSlide(){
  slide.setColorDepth(8);
  slide.createSprite(480, 240);
  slide.pushImage(0, 0, 480, 240, Slide);
}

**Comment by Bodmer** [comment, marked as answer]
loop() will keep cycling around so you need to set a limit on the number of times the sprite is scrolled and drawn. Try this:

```
void loop() {
  static uint16_t x_shift = 0;
  while (x_shift < 240) {
    slide.pushSprite(-240, 0, TFT_TRANSPARENT);
    slide.scroll(2);
    x_shift += 2;
  }
}
```

**Reply by whogate** [reply]
Worked perfect, thank you!
